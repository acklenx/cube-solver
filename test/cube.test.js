import {
  edgeOrientationMove,
} from '../src/cube';

test('perform an edge orientation move', () => {
  expect(edgeOrientationMove([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], 0))
    .toEqual([0, 1, 0, 0, 0, 1, 0, 0, 1, 1, 0, 0]);
});

test('perform an edge orientation move twice', () => {
  let eo = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0];

  for (let i = 0; i < 2; i += 1) {
    eo = edgeOrientationMove(eo, 0);
  }

  expect(eo).toEqual([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]);
});
