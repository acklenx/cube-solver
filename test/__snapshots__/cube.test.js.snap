// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cube performs a corner orientation move 1`] = `
Array [
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
]
`;

exports[`cube performs a corner permutation move 1`] = `
Array [
  1,
  5,
  2,
  3,
  0,
  4,
  6,
  7,
]
`;

exports[`cube performs an edge orientation move 1`] = `
Array [
  0,
  1,
  0,
  0,
  0,
  1,
  0,
  0,
  1,
  1,
  0,
  0,
]
`;

exports[`cube performs an edge permutation move 1`] = `
Array [
  0,
  9,
  2,
  3,
  4,
  8,
  6,
  7,
  1,
  5,
  10,
  11,
]
`;

exports[`cube resets the edge orientation when doing a move twice 1`] = `
Array [
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
]
`;
